cirq/ops/common_channels.py

```diff
@@ -184,6 +195,700 @@ def depolarize(p: float) -> DepolarizingChannel:
+                raise ValueError('{} was less than 0.'.format(p_str))
+            elif p > 1:
+                raise ValueError('{} was greater than 1.'.format(p_str))
+            return p
+
+        self._p = validate_probability(p, 'p')
+        self._delegate = AsymmetricDepolarizingChannel(1. - p, 0., 0.)
+
+    def _channel_(self) -> Iterable[np.ndarray]:
+        # Return just the I and X pieces.
+        return list(self._delegate._channel_())[:2]
+
+    def _eq_tuple(self):
+        return (BitFlipChannel, self._p)
+
+    def __eq__(self, other):
+        if not isinstance(other, type(self)):
+            return NotImplemented
+        return self._eq_tuple() == other._eq_tuple()
+
+    def __hash__(self):
+        return hash(self._eq_tuple())
+
+    def __ne__(self, other):
+        return not self == other
+
+    def __repr__(self) -> str:
+        return 'cirq.bit_flip(p={!r})'.format(self._p)
+
+    def __str__(self) -> str:
+        return 'bit_flip(p={!r})'.format(self._p)
+
+    def _circuit_diagram_info_(
+        self, args: protocols.CircuitDiagramInfoArgs
+    ) -> str:
+        return 'BF({!r})'.format(self._p)
+
+
+def _bit_flip_X() -> XPowGate:
+    """
+    Returns a cirq.X with guarantees a bit flip
+    """
+    return XPowGate()
+
+
+def _bit_flip(p: float) -> BitFlipChannel:
+    """
+    Construct a BitFlipChannel that flips a qubit state
+    with probability of a flip given by p.
+
+    This channel evolves a density matrix via
+            \rho -> M_0 \rho M_0^\dagger + M_1 \rho M_1^\dagger
+
+    With
+        M_0 = \sqrt{p} \begin{bmatrix}
+                            1 & 0 \\
+                            0 & 1
+                       \end{bmatrix}
+        M_1 = \sqrt{1-p} \begin{bmatrix}
+                            0 & 1 \\
+                            1 & -0
+                         \end{bmatrix}
+
+    Args:
+        p: the probability of a bit flip
+
+    Raises:
+        ValueError: if p is not a valid probability.
+    """
+    return BitFlipChannel(p)
+
+
+def bit_flip(p: Optional[float] = None) -> Union[XPowGate, BitFlipChannel]:
+    """
+    Construct a BitFlipChannel that flips a qubit state
+    with probability of a flip given by p. If p is None, return
+    a guaranteed flip in the form of an X operation.
+
+    This channel evolves a density matrix via
+            \rho -> M_0 \rho M_0^\dagger + M_1 \rho M_1^\dagger
+
+    With
+        M_0 = \sqrt{p} \begin{bmatrix}
+                            1 & 0 \\
+                            0 & 1
+                       \end{bmatrix}
+        M_1 = \sqrt{1-p} \begin{bmatrix}
+                            0 & 1 \\
+                            1 & -0
+                         \end{bmatrix}
+
+    Args:
+        p: the probability of a bit flip
+
+    Raises:
+        ValueError: if p is not a valid probability.
+    """
+    if p is None:
+        return _bit_flip_X()
+
+    return _bit_flip(p)
+
+
+class RotationErrorChannel(raw_types.Gate):
```

Comment:
I think we can design a more general version of this, but I'm fine with this being the starting point.

Reply:
sounds good to me haha